src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -510,56 +526,78 @@ private int getPossibleNewSlavesCount(SlaveTemplate template) throws AmazonClien
         try {
-            List<PlannedNode> r = new ArrayList<PlannedNode>();
+            List<PlannedNode> r = new ArrayList<>();
             final SlaveTemplate t = getTemplate(label);
             LOGGER.log(Level.INFO, "Attempting to provision slave from template " + t + " needed by excess workload of " + excessWorkload + " units of label '" + label + "'");
             if (label == null) {
                 LOGGER.log(Level.WARNING, String.format("Label is null - can't calculate how many executors slave will have. Using %s number of executors", t.getNumExecutors()));
             }
             while (excessWorkload > 0) {
-                final EC2AbstractSlave slave = getNewOrExistingAvailableSlave(t, label, false);
+
                 // Returned null if a new node could not be created
-                if (slave == null)
-                    break;
+//                if (slave == null)
+//                    break;
                 LOGGER.log(Level.INFO, String.format("We have now %s computers", Jenkins.getInstance().getComputers().length));
-                Jenkins.getInstance().addNode(slave);
-                LOGGER.log(Level.INFO, String.format("Added node named: %s, We have now %s computers", slave.getNodeName(), Jenkins.getInstance().getComputers().length));
+//                LOGGER.log(Level.INFO, String.format("Added node named: %s, We have now %s computers", slave.getNodeName(), Jenkins.getInstance().getComputers().length));
                 r.add(new PlannedNode(t.getDisplayName(), Computer.threadPoolForRemoting.submit(new Callable<Node>() {
 
                     public Node call() throws Exception {
-                        long startTime = System.currentTimeMillis(); // fetch starting time
-                        while ((System.currentTimeMillis() - startTime) < slave.launchTimeout * 1000) {
-                            return tryToCallSlave(slave, t);
-                        }
-                        LOGGER.log(Level.WARNING, "Expected - Instance - failed to connect within launch timeout");
-                        return tryToCallSlave(slave, t);
+                        final EC2AbstractSlave slave = getNewOrExistingAvailableSlave(t, label, false);
+//                        Jenkins.getInstance().addNode(slave);
+//                        long startTime = System.currentTimeMillis(); // fetch starting time
+//                        while ((System.currentTimeMillis() - startTime) < slave.launchTimeout * 1000) {
+//                            return tryToCallSlave(slave, t);
+//                        }
+
+//                        LOGGER.log(Level.WARNING, "Expected - Instance - failed to connect within launch timeout");
+//                        return tryToCallSlave(slave, t);
+                        Jenkins.getInstance().addNode(slave);
+                        slave.toComputer().connect(false).get();
+                        return slave;
```

Comment:
All the `call` method should do is to provision the slave, add the node to Jenkins and return the slave.